src/frontend/apps/desk/src/pages/404.tsx

```diff
@@ -1,25 +1,45 @@
+import { Button } from '@openfun/cunningham-react';
 import { ReactElement } from 'react';
 import { useTranslation } from 'react-i18next';
+import styled from 'styled-components';
 
-import { Text } from '@/components';
+import Icon404 from '@/assets/icons/icon-404.svg';
+import { Box, StyledLink, Text } from '@/components';
 import { NextPageWithLayout } from '@/types/next';
 
 import MainLayout from './MainLayout';
 
+const StyledButton = styled(Button)`
+  width: fit-content;
+  padding-left: 2rem;
+  padding-right: 2rem;
+`;
+
 const Page: NextPageWithLayout = () => {
   const { t } = useTranslation();
 
   return (
-    <Text
-      className="mt-xl"
-      as="h1"
-      $justify="center"
-      $align="center"
-      $theme="danger"
-      $textAlign="center"
-    >
-      {t('404 - Page not found')}
-    </Text>
+    <Box $align="center" className="m-auto" $height="70vh" $gap="2rem">
+      <Icon404 aria-label="Image 404" role="img" />
+
+      <Box $align="center">
+        <Text $size="h2" $weight="700" $theme="greyscale" $variation="900">
+          {t('Ouch !')}
+        </Text>
+      </Box>
+
+      <Text as="p" $textAlign="center" $css="max-width: 400px">
+        {t(
+          'It seems that the page you are looking for does not exist or cannot be displayed correctly.',
+        )}
+      </Text>
+
+      <Box $css="margin-top: 3rem">
```

Comment:
```suggestion
      <Box className="mt-l">
```

Reply:
For the spacings we try to use the Cunningham helpers:
https://openfun.github.io/cunningham/storybook/?path=/docs/getting-started-spacings--docs